docker-qgis/apply_deltas.py

```diff
@@ -500,7 +500,8 @@ def committed_features_added_cb(layer_id, features):
             if feature.isValid():
                 _pk_attr_idx, pk_attr_name = find_layer_pk(layer)
 
-                assert pk_attr_name
+                if not pk_attr_name:
+                    raise DeltaException(f"Layer {layer.name()} has not primary key")
```

Comment:
```suggestion
                    raise DeltaException(f'Layer "{layer.name()}" has no primary key.')
```
Even though not in place everywhere, I usually put placeholders in double quotes. Also, I am pretty sure it is "the layer has no primary key", but English is not my superpower.